roxctl/central/crs/generate.go

```diff
@@ -0,0 +1,108 @@
+package crs
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	pkgCommon "github.com/stackrox/rox/pkg/roxctl/common"
+	"github.com/stackrox/rox/pkg/stringutils"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+)
+
+// generateCRS generates a new CRS using Central's API and writes the newly generated CRS into the
+// file specified by `outFilename` (if it is non-empty) or to stdout (if `outFilename` is empty).
+func generateCRS(cliEnvironment environment.Environment, name string,
+	outFilename string, timeout time.Duration, retryTimeout time.Duration,
+) error {
+	var err error
+	var outFile *os.File
+
+	ctx, cancel := context.WithTimeout(pkgCommon.Context(), timeout)
+	defer cancel()
+
+	conn, err := cliEnvironment.GRPCConnection(common.WithRetryTimeout(retryTimeout))
+	if err != nil {
+		return err
+	}
+	defer utils.IgnoreError(conn.Close)
+	svc := v1.NewClusterInitServiceClient(conn)
+
+	outWriter := cliEnvironment.InputOutput().Out()
+	if outFilename != "" {
+		outFile, err = os.OpenFile(outFilename, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0600)
+		if err != nil {
+			return errors.Wrap(err, "creating output file for writing CRS")
+		}
+		outWriter = outFile
+		defer func() {
+			_ = outFile.Close()
+			if err != nil {
+				utils.Should(os.Remove(outFilename))
+			}
+		}()
+	}
+
+	req := v1.CRSGenRequest{Name: name}
+	resp, err := svc.GenerateCRS(ctx, &req)
+	if err != nil {
```

Comment:
nitpick (just in case you want to go a step further): it would be great if we could distinguish between generation failed and central does not support that functionality. And provide better feedback to users.

Reply:
done